android-project/app/src/main/java/org/diasurgical/devilutionx/DevilutionXSDLActivity.java

```diff
@@ -1,136 +1,152 @@
-		new AlertDialog.Builder(this)
-				.setTitle(SharewareTitle)
-				.setPositiveButton("OK", null)
-				.setMessage(message)
-				.show();
-		installationGuideShown();
-	}
-
-	private void triggerPulse() {
-		permissionHandler.postDelayed(new Runnable() {
-			@Override
-			public void run() {
-				if (PackageManager.PERMISSION_GRANTED == checkSelfPermission(permission)) {
-					PermissionDialog.hide();
-					throw new RuntimeException();
-				}
-				triggerPulse();
-			}
-		}, 50);
-	}
-
-	private boolean hasDataFile() {
-		File fileLower = new File(System.getenv("EXTERNAL_STORAGE") + "/diabdat.mpq");
-		File fileUpper = new File(System.getenv("EXTERNAL_STORAGE") + "/DIABDAT.MPQ");
-
-		return fileUpper.exists() || fileLower.exists();
-	}
-
-	protected String[] getLibraries() {
-		return new String[]{
-				"SDL2",
-				"SDL2_ttf",
-				"devilutionx"
-		};
-	}
-}
+package org.diasurgical.devilutionx;
+
+import android.Manifest;
+import android.app.AlertDialog;
+import android.content.SharedPreferences;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.Environment;
+import android.preference.PreferenceManager;
+
+import org.libsdl.app.SDLActivity;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+public class DevilutionXSDLActivity extends SDLActivity {
+	private final String SharewareTitle = "Shareware version";
+	private SharedPreferences mPrefs;
+	final private String shouldShowInstallationGuidePref = "shouldShowInstallationGuide";
+	String externalDir;
+
+	protected void onCreate(Bundle savedInstanceState) {
+		externalDir = getExternalFilesDir(null).getAbsolutePath();
+
+		migrateAppData();
+
+		if (shouldShowInstallationGuide()) {
+			showInstallationGuide();
+		}
+
+		super.onCreate(savedInstanceState);
+	}
+
+	/**
+	 * Check if this is the first time the app is run and the full game data is not already present
+	 */
+	private boolean shouldShowInstallationGuide() {
+		mPrefs = PreferenceManager.getDefaultSharedPreferences(this);
+		return mPrefs.getBoolean(shouldShowInstallationGuidePref, !hasFullGameData());
+	}
+
+	private boolean hasFullGameData() {
+		File fileLower = new File(externalDir + "/diabdat.mpq");
+		File fileUpper = new File(externalDir + "/DIABDAT.MPQ");
+
+		return fileUpper.exists() || fileLower.exists();
+	}
+
+	private void installationGuideShown() {
+		SharedPreferences.Editor editor = mPrefs.edit();
+		editor.putBoolean(shouldShowInstallationGuidePref, false);
+		editor.apply();
+	}
+
+	private void showInstallationGuide() {
+		String message = "To play the full version you must place DIABDAT.MPQ from the original game in Android"
+				+ externalDir.split("/Android")[1]

```

Comment:
The path will usually be something like `/storage/Emulated/0/Android/data/org.diasurgical.devilutionx/files`, but `/storage/Emulated/<int>` or (`/storage/<UUID>`) is never exposed via the UI or when connecting via USB, instead they would be shown as `SD`, `Internal Storage`, `Phone` or `Card` (depending on the tool used). So showing the actual path would be unhelpful to most users, possibly leading them to place the files in `/storage/Emulated/0/storage/Emulated/0/Android/data/org.diasurgical.devilutionx/files` instead of the intended folder.